### PR TITLE
HIVE-25224: Multi insert statements involving tables with different bucketing_versions results in error

### PR DESCRIPTION
**Pull request by kgyrtkirk**

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/BucketVersionPopulator.java [hunk 1]
```diff
@@ -131,22 +132,47 @@ private OpGroup newGroup(NodeProcessorCtx procCtx) {
...
+        new Comparator<BucketVersionPopulator.OperatorBucketingVersionInfo>() {
 
+          @Override
+          public int compare(OperatorBucketingVersionInfo i1, OperatorBucketingVersionInfo i2) {
+            int r = i1.infoType.compareTo(i2.infoType);
+            if (r != 0) {
+              // mandatory first
+              return r;
+            }
+            r = i2.bucketingVersion - i1.bucketingVersion;
```
Comment (kasakrisz) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/BucketVersionPopulator.java:154]: nit. `Integer.compare(i2.bucketingVersion, i1.bucketingVersion)` can be used here.

**Review by kasakrisz — APPROVED** [review]
LGTM

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/BucketVersionPopulator.java (hunk 1 above)
Comment (kgyrtkirk) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/BucketVersionPopulator.java]: :+1:

**Comment by kgyrtkirk** [comment]
I've merged this - testruns had multiple unrelated  failures in a row; I'll go and clean up these flaky tests....

**Pull request merged**